platform/android/app/src/test/kotlin/coop/polypoly/polypod/polyout/PolyOutTest.kt

```diff
@@ -0,0 +1,298 @@
+            stream: OutputStream?,
+            password: CharArray?
+        ) = wrapped.store(stream, password)
+
+        override fun engineSize(): Int = wrapped.size()
+
+        override fun engineAliases(): Enumeration<String> = Collections.enumeration(storedKeys.keys) // ktlint-disable max-line-length
+
+        override fun engineContainsAlias(
+            alias: String?
+        ): Boolean = storedKeys.containsKey(alias)
+
+        override fun engineLoad(
+            stream: InputStream?,
+            password: CharArray?
+        ) = wrapped.load(stream, password)
+
+        override fun engineGetCertificateChain(
+            alias: String?
+        ): Array<Certificate>? = wrapped.getCertificateChain(alias)
+
+        override fun engineSetCertificateEntry(
+            alias: String?,
+            cert: Certificate?
+        ) = wrapped.setCertificateEntry(alias, cert)
+
+        override fun engineGetCertificateAlias(
+            cert: Certificate?
+        ): String? = wrapped.getCertificateAlias(cert)
+
+        override fun engineGetKey(
+            alias: String?,
+            password: CharArray?
+        ): Key? = (storedKeys[alias] as? KeyStore.SecretKeyEntry)?.secretKey
+
+        override fun engineGetEntry(
+            p0: String,
+            p1: KeyStore.ProtectionParameter?
+        ): KeyStore.Entry? = storedKeys[p0]
+
+        override fun engineSetEntry(
+            p0: String,
+            p1: KeyStore.Entry,
+            p2: KeyStore.ProtectionParameter?
+        ) {
+            storedKeys[p0] = p1
+        }
+
+        override fun engineLoad(
+            p0: KeyStore.LoadStoreParameter?
+        ) = wrapped.load(p0)
+
+        override fun engineStore(
+            p0: KeyStore.LoadStoreParameter?
+        ) = wrapped.store(p0)
+
+        override fun engineEntryInstanceOf(
+            p0: String?,
+            p1: Class<out KeyStore.Entry>?
+        ) = wrapped.entryInstanceOf(p0, p1)
+
+        companion object {
+            private val wrapped = KeyStore.getInstance("BKS", "BC")
+            internal val storedKeys = mutableMapOf<String, KeyStore.Entry>()
+        }
+    }
+
+    class FakeAesKeyGenerator : KeyGeneratorSpi() {
+        private val wrapped = KeyGenerator.getInstance("AES", "BC")
+        private var lastSpec: AlgorithmParameterSpec? = null
+
+        override fun engineInit(random: SecureRandom?) = wrapped.init(random)
+
+        override fun engineInit(
+            params: AlgorithmParameterSpec?,
+            random: SecureRandom?
+        ) = wrapped.init(random).also {
+            lastSpec = params
+        }
+
+        override fun engineInit(
+            keysize: Int,
+            random: SecureRandom?
+        ) = wrapped.init(keysize, random)
+
+        override fun engineGenerateKey(): SecretKey = wrapped.generateKey()
+    }
+
+    private var polyOut: PolyOut = PolyOut(
+        context = InstrumentationRegistry.getInstrumentation().targetContext
```

Comment:
Nope, context is not initialised yet.

Reply:
True, but Kotlin also has the concept of lazy variables.

Reply:
Is this the private var you're defining below? Is there any reason why it's not defined above?

Reply:
Easier to read the code with all the mocked classes above.

Reply:
Right, but if you finally follow @GhenadiePusca suggestion it would make sense in order to understand there's a local `context` variable that's assigned to that argument.